test_conformance/api/test_context_destructor_callback.cpp

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "testBase.h"
+
+static volatile cl_int sDestructorIndex;
+
+void CL_CALLBACK context_destructor_callback(cl_context context, void *userData)
+{
+    int *userPtr = (int *)userData;
+
+    // ordering of callbacks is guaranteed, meaning we don't need to do atomic
+    // operation here
+    *userPtr = ++sDestructorIndex;
+}
+
+#ifndef ABS
+#define ABS(x) ((x < 0) ? -x : x)
+#endif
+
+int test_context_destructor_callback(cl_device_id deviceID, cl_context context,
+                                     cl_command_queue queue, int num_elements)
+{
+    cl_int error;
+    clContextWrapper localContext =
+        clCreateContext(NULL, 1, &deviceID, NULL, NULL, &error);
+    test_error(error, "Unable to create local context");
+
+    int i;
```

Comment:
Maybe better to declare it as part of the for loop below.

Reply:
Done!